### PR TITLE
Do not restore the database for a site if the site has skip_provisioning as true

### PR DESCRIPTION
**Pull request by Mte90**
## Summary:

Fix https://github.com/Varying-Vagrant-Vagrants/VVV/issues/2131 about the database side.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ permalink: /docs/en-US/changelog/
 * Always set the database root user password to avoid having the default invalid password on fresh installs
 * Don't spider and recurse domains when checking for network connections
 * Swap the MariaDB apt mirror used for a more reliable source
+* Not import the database if the site has `skip_provisioning` as true
```
Comment (msaggiorato) [inline CHANGELOG.md:30]: I'd say that the changelog would be "Do not restore the database for a site if the site has `skip_provisioning` as true"

File: database/sql/import-sql.sh [hunk 2]
```diff
@@ -48,6 +48,10 @@ then
 	pre_dot=${file%%.sql}
 	# get rid of the ./
   db_name=${pre_dot##./}
+	if [[ True == $(get_config_value "sites.${db_name}.skip_provisioning" "False") ]]; then
```
Comment (msaggiorato) [inline database/sql/import-sql.sh:51]: Wouldn't True need to be on quotes (since it's a string)?

**Comment by tomjn** [comment]
I don't believe it's possible to implement this, and I question wether we should. It assumes that a database is created for each site, and only that database. But that assumption isn't necessarily true

File: database/sql/import-sql.sh (hunk 2 above)
Comment (Mte90) [reply database/sql/import-sql.sh:51]: I copied that code from another provisioner.

**Comment by Mte90** [comment]
I have like 20 database backups in VVV of older webistes that not exists anymore, with this change I can confirm that is more fast the provisiong.
Also that some templates like the meta one that create like 4 db are not reconized, so they are imported anyway.

Looking at the original template there is also the `db_name` https://github.com/Varying-Vagrant-Vagrants/custom-site-template but this code doesn't use that value. I can add a check for this as first otherwise use the site name as fallback, so people with issue can change it.

**Comment by benlumia007** [comment]
i think the skip_provisioning also causing sites to create certifications if the skip is true.

**Comment by msaggiorato** [comment]
What happens if a database name doesn't match any entry in the config.yml file? Would it be imported?

This will still allow the flow that @tomjn mentions of people potentially having 2 databases related to a site.

**Comment by Mte90** [comment]
Not imported, there is the `db_name` parameter, if it is defined it will be used that so if someone use a different db is better that configure it in the config file.

**Comment by msaggiorato** [comment]
But db_name currently doesn't allow a list, it's just a string right? So the use case that @tomjn mentioned which is usually possible with the current "dirty" implementation is not possible on the new "clean" one.

We need to somehow accomodate for this use case, but still avoid restoring things that should not be restored. See what I mean?

**Comment by Mte90** [comment]
Well if a template install and create more database they will be ignored on this process so they will be imported.

**Comment by tomjn** [comment]
The more I think about this, the more I think that we should provide general database back up options rather than site specific ones. Otherwise we'd be better off giving every site their own user+pass and set of databases which is a lot of work and doesn't help existing users

I'd suggest `config.yml` have options for:


 - back up specific database
 - skip specific database
 - default database backup value ( e.g. if it's not in the above two options what should we do?

Then again for restore, under a new `db_backup:` section.

E.g. something like this that covers all bases:

```yaml
general:
  db_backup:
    default: true
    include:
      - firstdb
      - second
    exclude:
      - fourth
      - fifth
  db_restore:
    default: true
    include:
      - firstdb
      - second
    exclude:
      - fourth
      - fifth
```

There's also the question of changing our default backup behaviour, e.g. only backup on `vagrant destroy`, and present manual restore options for each database.

**Comment by Mte90** [comment]
Closed for https://github.com/Varying-Vagrant-Vagrants/VVV/issues/2225

**Pull request closed without merging**